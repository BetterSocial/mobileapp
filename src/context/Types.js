export const ADD_CONTACT_LIST = 'ADD_CONTACT_LIST';
export const ADD_NEWS_I_FOLLOW = 'ADD_NEWS_I_FOLLOW';
export const NAVBAR_TITLE_MY_PROFILE = 'NAVBAR_TITLE_MY_PROFILE';
export const SET_ASSET_GROUP_CHAT = 'SET_ASSET_GROUP_CHAT';
export const SET_CHANNEL = 'SET_CHANNEL';
export const SET_CLIENT = 'SET_CLIENT';
export const SET_DATA_HUMAN_ID = 'SET_DATA_HUMAN_ID';
export const SET_DATA_IMAGE = 'SET_DATA_IMAGE';
export const SET_DATA_IMAGE_URL = 'SET_DATA_IMAGE_URL';
export const SET_DATA_USERNAME = 'SET_DATA_USERNAME';
export const SET_FEED = 'SET_FEED';
export const SET_FEED_BY_ID = 'SET_FEED_BY_ID';
export const SET_FEED_BY_INDEX = 'SET_FEED_BY_INDEX';
export const SET_LOCAL_COMUNITY = 'SET_LOCAL_COMUNITY';
export const SET_MY_PROFILE = 'SET_MY_PROFILE';
export const SET_MY_PROFILE_FEED = 'SET_MY_PROFILE_FEED';
export const SET_MY_PROFILE_FEED_BY_INDEX = 'SET_MY_PROFILE_FEED_BY_INDEX';
export const SET_NEWS = 'SET_NEWS';
export const SET_NEWS_I_FOLLOW = 'SET_NEWS_I_FOLLOW';
export const SET_OTHER_PROFILE_FEED = 'SET_OTHER_PROFILE_FEED';
export const SET_OTHER_PROFILE_FEED_BY_INDEX = 'SET_OTHER_PROFILE_FEED_BY_INDEX';
export const SET_PARTICIPANTS_GROUP_CHAT = 'SET_PARTICIPANTS_GROUP_CHAT';
export const SET_PROFILE_CHANNEL = 'SET_PROFILE_CHANNEL';
export const SET_TOPICS = 'SET_TOPICS';
export const SET_TOTAL_UN_READ_MESSAGE = 'SET_TOTAL_UN_READ_MESSAGE';
export const SET_UN_READ_MESSAGE = 'SET_UN_READ_MESSAGE';
export const SHOW_HEADER_NAVBAR = 'SHOW_HEADER_NAVBAR';
export const DISCOVERY_SET_FIRST_TIME_OPEN = 'DISCOVERY_SET_FIRST_TIME_OPEN';
export const DISCOVERY_SET_SEARCH_BAR = 'DISCOVERY_SET_SEARCH_BAR';
export const DISCOVERY_SET_DATA = 'DISCOVERY_SET_DATA';
export const DISCOVERY_SET_LOADING_DATA = 'DISCOVERY_SET_LOADING_DATA';
export const DISCOVERY_SET_NEW_FOLLOWED_USER = 'DISCOVERY_SET_NEW_FOLLOWED_USER';
export const DISCOVERY_SET_NEW_UNFOLLOWED_USER = 'DISCOVERY_SET_NEW_UNFOLLOWED_USER';
export const DISCOVERY_SET_NEW_FOLLOWED_DOMAIN = 'DISCOVERY_SET_NEW_FOLLOWED_DOMAIN';
export const DISCOVERY_SET_NEW_UNFOLLOWED_DOMAIN = 'DISCOVERY_SET_NEW_UNFOLLOWED_DOMAIN';
export const SET_SELECTED_LAST_DOMAIN = 'SET_SELECTED_LAST_DOMAIN';
export const SET_DOMAIN_DATA = 'SET_DOMAIN_DATA';
export const SET_PROFILE_DOMAIN = 'SET_PROFILE_DOMAIN';
export const SET_FOLLOWING_USER = 'SET_FOLLOWING_USER';
export const SET_FOLLOWING_DOMAIN = 'SET_FOLLOWING_DOMAIN';
export const SET_FOLLOWING_TOPICS = 'SET_FOLLOWING_TOPIC';
