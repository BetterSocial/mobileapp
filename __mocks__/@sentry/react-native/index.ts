export const addBreadcrumb = jest.fn();
