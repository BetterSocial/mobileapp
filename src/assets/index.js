import homeSelected from './icon/HomeSelected';
import homeUnselected from './icon/HomeUnselected';
import feedSelected from './icon/FeedSelected';
import feedUnselected from './icon/FeedUnselected';
import newsSelected from './icon/NewsSelected';
import newsUnselected from './icon/NewsUnselected';
import profileSelected from './icon/ProfileSelected';
import profileUnselected from './icon/ProfileUnselected';
import AllowLeft from './arrow-left.png';

const homeTabs = {
  homeSelected,
  homeUnselected,
  feedSelected,
  feedUnselected,
  newsSelected,
  newsUnselected,
  profileSelected,
  profileUnselected,
  AllowLeft
};

export {homeTabs};
