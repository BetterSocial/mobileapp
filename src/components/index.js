import Avatar from './Avatar';
import SingleSidedShadowBox from './SingleSidedShadowBox';
import Footer from './Footer/Footer';

export {Avatar, SingleSidedShadowBox, Footer};
