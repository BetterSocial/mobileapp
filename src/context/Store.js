import * as React from 'react';

import {channelReducer, channelState} from './reducers/channelReducer';
import {clientReducer, clientState} from './reducers/clientReducer';
import {discoveryReducer, discoveryState} from './reducers/discoveryReducer';
import {domainReducer, domainState} from './reducers/domainReducer';
import {feedsReducer, feedsState} from './reducers/FeedReducer';
import {followingReducer, followingState} from './reducers/followingReducer';
import {generalComponentReducer, generalComponentState} from './reducers/generalComponentReducer';
import {groupChatReducer, groupChatState} from './reducers/groupChat';
import {initialStartupReducer, initialStartupState} from './reducers/intialStartupReducer';
import {localCommunityReducer, localCommunityState} from './reducers/localComunityReducer';
import {myProfileFeedReducer, myProfileFeedState} from './reducers/myProfileFeedReducer';
import {myProfileReducer, myProfileState} from './reducers/myProfileReducer';
import {newsReducer, newsState} from './reducers/newsReducer';
import {otherProfileFeedReducer, otherProfileFeedState} from './reducers/OtherProfileFeedReducer';
import {topicsReducer, topicsState} from './reducers/topicsReducer';
import {unReadMessageReducer, unReadMessageState} from './reducers/unReadMessageReducer';
import {usersReducer, usersState} from './reducers/userReducer';
import {commentReducer, commentState} from './reducers/commentReducer';

/**
 *
 * @typedef {Object} MainContext
 * @property {import('./reducers/discoveryReducer').DiscoveryState} discovery
 */
const Store = ({children}) => {
  const rootReducer = {
    channel: React.useReducer(channelReducer, channelState),
    client: React.useReducer(clientReducer, clientState),
    discovery: React.useReducer(discoveryReducer, discoveryState),
    feeds: React.useReducer(feedsReducer, feedsState),
    following: React.useReducer(followingReducer, followingState),
    generalComponent: React.useReducer(generalComponentReducer, generalComponentState),
    groupChat: React.useReducer(groupChatReducer, groupChatState),
    localCommunity: React.useReducer(localCommunityReducer, localCommunityState),
    myProfileFeed: React.useReducer(myProfileFeedReducer, myProfileFeedState),
    otherProfileFeed: React.useReducer(otherProfileFeedReducer, otherProfileFeedState),
    news: React.useReducer(newsReducer, newsState),
    profile: React.useReducer(myProfileReducer, myProfileState),
    topics: React.useReducer(topicsReducer, topicsState),
    unReadMessage: React.useReducer(unReadMessageReducer, unReadMessageState),
    users: React.useReducer(usersReducer, usersState),
    domains: React.useReducer(domainReducer, domainState),
    initialStartup: React.useReducer(initialStartupReducer, initialStartupState),
    comments: React.useReducer(commentReducer, commentState)
  };
  return <Context.Provider value={rootReducer}>{children}</Context.Provider>;
};
export const Context = React.createContext(null);
export const useAppContext = () => React.useContext(Context);

export default Store;
