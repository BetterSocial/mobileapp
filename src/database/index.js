import SQLite from 'react-native-sqlite-storage';

SQLite.enablePromise(true);
const DB_NAME = 'bettersocial.db';

let db = null;

const getDBConnection = async () => {
  if (db) return Promise.resolve(db);
  db = SQLite.openDatabase(
    {
      name: DB_NAME,
      location: 'default'
    },
    () => {
      console.log('success connecting database');
    },
    (e) => {
      console.log('failed connecting database');
      console.log(e);
      return Promise.reject(e);
    }
  );

  return Promise.resolve(db);
};

const LocalDatabase = {
  getDBConnection
};

export default LocalDatabase;
