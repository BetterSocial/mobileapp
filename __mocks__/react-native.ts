import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN