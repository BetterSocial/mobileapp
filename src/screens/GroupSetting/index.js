import * as React from 'react';
import {
  Dimensions,
  FlatList,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  View
} from 'react-native';

import ButtonAddParticipants from '../../components/Button/ButtonAddParticipants';
import EditGroup from './elements/EditGroup';
import HeaderContact from '../../components/Header/HeaderContact';
import Loading from '../Loading';
import useGroupSetting from './hooks/useGroupSetting';
import {COLORS} from '../../utils/theme';
import {ProfileContact} from '../../components/Items';
import {fonts} from '../../utils/fonts';

const {width} = Dimensions.get('screen');

const GroupSetting = ({navigation, route}) => {
  const {
    participants,
    groupName,
    countUser,
    urlImage,
    isLoading,
    updateName,
    submitData,
    lounchGalery,
    renderHeaderSubtitleText
  } = useGroupSetting({navigation, route});

  const isFocusChatName = route?.params?.focusChatName;
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar translucent={false} />
      <ScrollView>
        <HeaderContact
          title={'Settings'}
          containerStyle={styles.containerHeader}
          subTitle={renderHeaderSubtitleText()}
          subtitleStyle={styles.subtitleStyle}
          onPressSub={submitData}
          onPress={() => navigation.goBack()}
        />
        <EditGroup
          imageUri={urlImage}
          editName={groupName}
          setEditName={updateName}
          onUpdateImage={lounchGalery}
          isFocusChatName={isFocusChatName}
          saveGroupName={() => submitData(false, false)}
        />
        <Loading visible={isLoading} />
        <View style={styles.users}>
          <Text style={styles.countUser}>{`Participants (${countUser})`}</Text>
          <FlatList
            data={Object.keys(participants)}
            keyExtractor={(item, index) => index.toString()}
            renderItem={({item}) => (
              <View style={{height: 72}}>
                <ProfileContact
                  key={item}
                  fullname={participants[item].user.name}
                  photo={participants[item].user.image}
                />
              </View>
            )}
          />
        </View>
        <ButtonAddParticipants />
      </ScrollView>
    </SafeAreaView>
  );
};

export default GroupSetting;

const styles = StyleSheet.create({
  container: {flex: 1, backgroundColor: '#fff'},
  subtitleStyle: {
    color: COLORS.holyTosca
  },
  containerHeader: {marginLeft: 22, marginRight: 20},
  users: {
    paddingTop: 12,
    width
  },
  countUser: {
    fontFamily: fonts.inter[600],
    fontSize: 14,
    lineHeight: 16.94,
    color: COLORS.holytosca,
    marginLeft: 20,
    marginBottom: 4
  }
});
