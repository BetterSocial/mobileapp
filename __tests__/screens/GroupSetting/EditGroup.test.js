import React from 'react'
import GroupSetting from "../../../src/screens/GroupSetting/elements/EditGroup"
import {render, cleanup, act, fireEvent} from '@testing-library/react-native'
import { Context } from '../../../src/context'

describe('EditGroup should run correctly', () => {
    const mockContext = {
            groupChat: [
                {asset: [
                    {message: {
                                    cid: "messaging:c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                                    created_at: "2023-01-24T00:59:12.801526Z",
                                    html: "",
                                    id: "c6c91b04-795c-404e-b012-ea28813a2006-531b41e6-263b-4d6c-1c0e-e62f13357aef",
                                    latest_reactions: [],
                                    mentioned_users: [],
                                    attachments: null,
                                    channel: {
                                        cid: "messaging:c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                                        created_at: "2022-09-30T22:49:45.59342Z",
                                        disabled: false,
                                        frozen: false,
                                        id: "c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                                        last_message_at: "2023-01-24T01:00:59.432027Z",
                                        member_count: 4,
                                        name: "Test group baru",
                                        type: "messaging",
                                    updated_at: "2023-01-24T01:41:46.237211Z"
                }
                    }}
                ],
                participants: {
                     'a3c59170-c110-4fac-929e-7834f6c6827f': {
                        banned: false,
                        channel_role: "channel_moderator",
                        created_at: "2022-09-30T22:49:45.911054Z",
                        is_moderator: true,
                        role: "admin",
                        shadow_banned: false,
                        updated_at: "2022-09-30T22:49:45.911054Z",
                        user: {
                            banned: false,
                            created_at: "2021-11-29T05:40:40.828927Z",
                            id: "a3c59170-c110-4fac-929e-7834f6c6827f",
                            image: "https://res.cloudinary.com/hpjivutj2/image/upload/v1666664085/vcinqpjniuigf6mdnmzz.jpg",
                            last_active: "2021-12-06T03:54:03.677683Z",
                            name: "BetterSocial_Team",
                            online: false,
                            role: "admin",
                            updated_at: "2022-11-02T15:30:30.170297Z"
                        },
                        user_id: "a3c59170-c110-4fac-929e-7834f6c6827f"
                    },
                        'b3c59170-c110-4fac-929e-7834f6c6827e': {
                        banned: false,
                        channel_role: "channel_moderator",
                        created_at: "2022-09-30T22:49:45.911054Z",
                        is_moderator: true,
                        role: "admin",
                        shadow_banned: false,
                        updated_at: "2022-09-30T22:49:45.911054Z",
                        user: {
                            banned: false,
                            created_at: "2021-11-29T05:40:40.828927Z",
                            id: "a3c59170-c110-4fac-929e-7834f6c6827f",
                            image: "https://res.cloudinary.com/hpjivutj2/image/upload/v1666664085/vcinqpjniuigf6mdnmzz.jpg",
                            last_active: "2021-12-06T03:54:03.677683Z",
                            name: "BetterSocial_Team",
                            online: false,
                            role: "admin",
                            updated_at: "2022-11-02T15:30:30.170297Z"
                        },
                        user_id: "b3c59170-c110-4fac-929e-7834f6c6827e"
                    },
                    'bc3c59170-c110-4fac-929e-7834f6c6827l': {
                        banned: false,
                        channel_role: "channel_moderator",
                        created_at: "2022-09-30T22:49:45.911054Z",
                        is_moderator: true,
                        role: "admin",
                        shadow_banned: false,
                        updated_at: "2022-09-30T22:49:45.911054Z",
                        user: {
                            banned: false,
                            created_at: "2021-11-29T05:40:40.828927Z",
                            id: "a3c59170-c110-4fac-929e-7834f6c6827f",
                            image: "https://res.cloudinary.com/hpjivutj2/image/upload/v1666664085/vcinqpjniuigf6mdnmzz.jpg",
                            last_active: "2021-12-06T03:54:03.677683Z",
                            name: "BetterSocial_Team",
                            online: false,
                            role: "admin",
                            updated_at: "2022-11-02T15:30:30.170297Z"
                        },
                        user_id: "bc3c59170-c110-4fac-929e-7834f6c6827l"
                    },
                    
                }
            }
            ],
            channel: [
                {channel: {
                                cid: "messaging:c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                    data: {
                        cid: "messaging:c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                        created_at: "2022-09-30T22:49:45.59342Z",
                        createdBy: {
                            banned: false,
                            created_at: "2022-06-10T13:11:53.396427Z",
                            id: "c6c91b04-795c-404e-b012-ea28813a2006",
                            image: "https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg",
                            last_active: "2022-06-10T13:11:58.020555Z",
                            name: "Agita",
                            online: false,
                            role: "user",
                            updated_at: "2023-01-24T01:41:19.021868Z",
                        },
                        disabled: false,
                        frozen: false,
                        hidden: false,
                    id: "c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                    last_message_at: "2023-01-24T01:00:59.432027Z",
                    member_count: 4,
                    name: "Test group baru",
                    type: "messaging",
                    updated_at: "2023-01-24T01:41:46.237211Z",
                    image: 'https://image.jpg'
                    },
                    disconnected: false,
                    id: "c47d45f2-0dd9-4eaa-1600-4ff6e518199a",
                    initialized: true,
                    isTyping: false,
                    lastKeyStroke: undefined,
                    lastTypingEvent: null,
                        },
                    profileChannel: null
                    }
            ],
            profile: [
                {myProfile: {
                      bio: "Fe mobile engineer",
        country_code: "ID",
        createdAt: "2022-06-10T13:11:53.000Z",
        follower: 13,
        follower_symbol: "< 25",
        following: 10,
        following_symbol: "< 25",
        human_id: "I4K3M10FGR78EWQQDNQ2",
        last_active_at: "2022-06-10T13:11:53.000Z",
        location: [
            {
                city: "Yauco, PR",
                country: "US",
                createdAt: "2022-05-30T14:15:24.000Z",
                location_id: "45",
                location_level: "City",
                neighborhood: "",
                slug_name: "",
                state: "Puerto Rico",
                status: "Y",
                updatedAt: "2022-05-30T14:15:24.000Z",
            }
        ],
        profile_pic_asset_id: "6f47f70bea98469f4a24b6ffc550c983",
        profile_pic_path: "https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg",
        profile_pic_public_id: "pbdv3jlyd4mhmtis6kqx",
        real_name: null,
        status: "Y",
        updatedAt: "2022-07-29T12:54:04.000Z",
        user_id: "c6c91b04-795c-404e-b012-ea28813a2006",
        username: "Agita",
                }}
            ]
        }
        const mockFocus = jest.fn()
        beforeEach(() => {
            jest.spyOn(React, 'useRef').mockImplementation(() => ({current: {focus: mockFocus}}))
        })

        it('editGroup should match snapshot', () => {
            const wrapper = ({children}) => (
            <Context.Provider value={mockContext} >
                {children}
            </Context.Provider>
            )
            const onUpdateImage = jest.fn()
            const saveGroupName = jest.fn()
            const updateName = jest.fn()
            const {toJSON} = render(<GroupSetting imageUri={'https://imag.jpg'} onUpdateImage={onUpdateImage} setEditName={updateName} editName={'general'} isFocusChatName={'test'} saveGroupName={saveGroupName} />, {wrapper})
            expect(toJSON).toMatchSnapshot()
        })

        it('updateImage should run correctly', () => {
             const wrapper = ({children}) => (
            <Context.Provider value={mockContext} >
                {children}
            </Context.Provider>
            )
            const onUpdateImage = jest.fn()
            const saveGroupName = jest.fn()
            const updateName = jest.fn()
            const {getByTestId} = render(<GroupSetting imageUri={'https://imag.jpg'} onUpdateImage={onUpdateImage} setEditName={updateName} editName={'general'} isFocusChatName={'test'} saveGroupName={saveGroupName} />, {wrapper})
            act(() => {
                fireEvent.press(getByTestId('updateImage'))
            })
            expect(onUpdateImage).toHaveBeenCalled()
        })

        it('image type file:/// should show correctly', () => {
               const wrapper = ({children}) => (
            <Context.Provider value={mockContext} >
                {children}
            </Context.Provider>
            )
            const onUpdateImage = jest.fn()
            const saveGroupName = jest.fn()
            const updateName = jest.fn()
            const {getAllByTestId} = render(<GroupSetting imageUri={'file:///imag.jpg'} onUpdateImage={onUpdateImage} setEditName={updateName} editName={'general'} isFocusChatName={'test'} saveGroupName={saveGroupName} />, {wrapper})
            expect(getAllByTestId('fileImage')).toHaveLength(1)
        })
         

            it('image type res.cloudinary.com should show correctly', () => {
               const wrapper = ({children}) => (
            <Context.Provider value={mockContext} >
                {children}
            </Context.Provider>
            )
            const onUpdateImage = jest.fn()
            const saveGroupName = jest.fn()
            const updateName = jest.fn()
            const {getAllByTestId} = render(<GroupSetting imageUri={'res.cloudinary.com'} onUpdateImage={onUpdateImage} setEditName={updateName} editName={'general'} isFocusChatName={'test'} saveGroupName={saveGroupName} />, {wrapper})
            expect(getAllByTestId('rescloud')).toHaveLength(1)
        })

        it('image count user > 2 should be correct', () => {
            const wrapper = ({children}) => (
            <Context.Provider value={mockContext} >
                {children}
            </Context.Provider>
            )
            const onUpdateImage = jest.fn()
            const saveGroupName = jest.fn()
            const updateName = jest.fn()
            const {getAllByTestId} = render(<GroupSetting imageUri={''} onUpdateImage={onUpdateImage} setEditName={updateName} editName={'general'} isFocusChatName={'test'} saveGroupName={saveGroupName} />, {wrapper})
            expect(getAllByTestId('imageCountUser')).toHaveLength(1)
        })
        
    
})