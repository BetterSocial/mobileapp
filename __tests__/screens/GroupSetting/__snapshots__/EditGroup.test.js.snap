// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGroup should run correctly editGroup should match snapshot 1`] = `[Function]`;
